python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/asset_daemon_scenario.py

```diff
@@ -206,33 +209,17 @@ class MultiAssetSpec(NamedTuple):
     can_subset: bool = False
 
 
-class AssetDaemonScenarioState(NamedTuple):
-    """Specifies the state of a given AssetDaemonScenario. This state can be modified by changing
-    the set of asset definitions it contains, executing runs, updating the time, evaluating ticks, etc.
-
-    At any point in time, assertions can be made about the state of the scenario. Typically, you
-    would add runs to the scenario, evaluate a tick, then make assertions about the runs that were
-    requested for that tick, or the evaluations that were stored for each asset.
-
-    Args:
-        asset_specs (Sequence[AssetSpec]): The specs describing all assets that are part of this
-            scenario.
-        current_time (datetime): The current time of the scenario.
+@dataclass(frozen=True)
+class AssetScenarioState:
```

Comment:
1) What is the contract between this base class and the derived class
2) When do we expect in the framework to be operating against a generic `AssetScenarioState` versus a `AssetDaemonScenarioState` or `AssetConditionScenarioState`

Just pressure testing whether inheritance is really the right thing here.

Reply:
@schrockn I think the cleaner mental model here is that both `AssetDaemonScenarioState` and `AssetConditionScenarioState` _have_ an `AssetScenarioState` which they can operate over. 

I actually originally implemented it this way but abandoned it for ergonomic reasons. Specifically, I felt like you'd either need to force the user to constantly punch through the abstraction (`daemon_state.with_asset_state(...)`) or re-implement all the methods on AssetScenarioState:

```python
class AssetDaemonScenarioState:
    ...

    def with_runs(self, runs):
         return self.with_asset_state(self.asset_state.with_runs(runs))
```

The other way around this would be to have some sort of `AssetStateHaver` base class which the `Daemon...State` and `Condition...State` classes both inherit from to at least avoid that bit of duping. Thoughts on that approach?